<launch>
  <!-- rqt_reconfigure -->
  <arg name="obstacleAvoidance/rqt_reconfigure" default="true"/>
  <!-- ロボットパラメータ -->
  <arg name="obstacleAvoidance/WheelD" default="0.3125"/>
  <arg name="obstacleAvoidance/angleMin" default="45"/>
  <arg name="obstacleAvoidance/angleMax" default="135"/>
  <arg name="obstacleAvoidance/angleDiv" default="1.0"/>
  <!-- ゴール位置 -->
  <arg name="obstacleAvoidance/goalX" default="0.0"/>
  <arg name="obstacleAvoidance/goalY" default="6.0"/>
  <!-- vfh -->
  <!-- vfh - k -->
  <arg name="obstacleAvoidance/marginRadius" default="1.0"/>
  <arg name="obstacleAvoidance/Kcp" default="0.5"/>
  <arg name="obstacleAvoidance/Kg" default="0.5"/>
  <arg name="obstacleAvoidance/KcurAngle" default="0.5"/>
  <arg name="obstacleAvoidance/KprevAngle" default="0.5"/>
  <!-- vfh - eta -->
  <arg name="obstacleAvoidance/EtaCp" default="1"/>
  <arg name="obstacleAvoidance/EtaG" default="1"/>
  <arg name="obstacleAvoidance/EtaCurAngle" default="1"/>
  <arg name="obstacleAvoidance/EtaPrevAngle" default="1"/>
  <!-- debug -->
  <arg name="obstacleAvoidance/debugType" default="1"/>
  <arg name="robot_name" default="robot2"/>
  <group ns="$(arg robot_name)">
  <node pkg="local_navigation" name="local_navigation_cp_vfh" type="local_navigation_cp_vfh" output="screen" respawn="true">
    <!-- rqt_reconfigure -->
    <param name="obstacleAvoidance/rqt_reconfigure" value="$(arg obstacleAvoidance/rqt_reconfigure)"/>
    <!-- ロボットパラメータ -->
    <param name="obstacleAvoidance/WheelD" value="$(arg obstacleAvoidance/WheelD)"/>
    <param name="obstacleAvoidance/angleMin" value="$(arg obstacleAvoidance/angleMin)"/>
    <param name="obstacleAvoidance/angleMax" value="$(arg obstacleAvoidance/angleMax)"/>
    <param name="obstacleAvoidance/angleDiv" value="$(arg obstacleAvoidance/angleDiv)"/>
    <!-- ゴール位置 -->
    <param name="obstacleAvoidance/goalX" value="$(arg obstacleAvoidance/goalX)"/>
    <param name="obstacleAvoidance/goalY" value="$(arg obstacleAvoidance/goalY)"/>
    <!-- vfh -->
    <!-- vfh - k -->
    <param name="obstacleAvoidance/marginRadius" value="$(arg obstacleAvoidance/marginRadius)"/>
    <param name="obstacleAvoidance/Kcp" value="$(arg obstacleAvoidance/Kcp)"/>
    <param name="obstacleAvoidance/Kg" value="$(arg obstacleAvoidance/Kg)"/>
    <param name="obstacleAvoidance/KcurAngle" value="$(arg obstacleAvoidance/KcurAngle)"/>
    <param name="obstacleAvoidance/KprevAngle" value="$(arg obstacleAvoidance/KprevAngle)"/>
    <!-- vfh - eta -->
    <param name="obstacleAvoidance/EtaCp" value="$(arg obstacleAvoidance/EtaCp)"/>
    <param name="obstacleAvoidance/EtaG" value="$(arg obstacleAvoidance/EtaG)"/>
    <param name="obstacleAvoidance/EtaCurAngle" value="$(arg obstacleAvoidance/EtaCurAngle)"/>
    <param name="obstacleAvoidance/EtaPrevAngle" value="$(arg obstacleAvoidance/EtaPrevAngle)"/>
    <!-- debug -->
    <param name="obstacleAvoidance/debugType" value="$(arg obstacleAvoidance/debugType)"/>
  </node>
  </group>
</launch>